src/python-common/ceph/tests/test_disk_selector.py

```diff
@@ -0,0 +1,760 @@
+import pytest
+
+from ceph.deployment.inventory import Devices, Device
+
+from ceph.deployment.drive_group import DriveGroupSpec, DriveGroupValidationError, DeviceSelection
+
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch  # type: ignore
+
+from ceph.deployment import drive_selection
+from ceph.tests.factories import InventoryFactory
+
+
+class TestMatcher(object):
+    """ Test Matcher base class
+    """
+
+    def test_get_disk_key_3(self):
+        """
+        virtual is False
+        key is found
+        retrun value of key is expected
+        """
+        disk_map = Device(path='/dev/vdb', sys_api={'foo': 'bar'})
+        ret = drive_selection.Matcher('foo', 'bar')._get_disk_key(disk_map)
+        assert ret is disk_map.sys_api.get('foo')
+
+    def test_get_disk_key_4(self):
+        """
+        virtual is False
+        key is not found
+        expect raise Exception
+        """
+        disk_map = Device(path='/dev/vdb')
+        with pytest.raises(Exception):
+            drive_selection.Matcher('bar', 'foo')._get_disk_key(disk_map)
+            pytest.fail("No disk_key found for foo or None")
+
+
+class TestSubstringMatcher(object):
+    def test_compare(self):
+        disk_dict = Device(path='/dev/vdb', sys_api=dict(model='samsung'))
+        matcher = drive_selection.SubstringMatcher('model', 'samsung')
+        ret = matcher.compare(disk_dict)
+        assert ret is True
+
+    def test_compare_false(self):
+        disk_dict = Device(path='/dev/vdb', sys_api=dict(model='nothing_matching'))
+        matcher = drive_selection.SubstringMatcher('model', 'samsung')
+        ret = matcher.compare(disk_dict)
+        assert ret is False
+
+
+class TestEqualityMatcher(object):
+    def test_compare(self):
+        disk_dict = Device(path='/dev/vdb', sys_api=dict(rotates='1'))
+        matcher = drive_selection.EqualityMatcher('rotates', '1')
+        ret = matcher.compare(disk_dict)
+        assert ret is True
+
+    def test_compare_false(self):
+        disk_dict = Device(path='/dev/vdb', sys_api=dict(rotates='1'))
```

Comment:
The attribute is renamed to `rotational`.

Reply:
Right, the way EqualityMatcher os implemented, this could be any string.